components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -361,183 +322,194 @@ void Nudging::run_impl (const double dt)
+
+    const int icol = team.league_rank();
+
+    // Setup
+    if (src_pmid_3d) {
+      vert_interp.setup(team, ekat::subview(p_mid_tmp_3d,icol),
+                              ekat::subview(p_tgt,icol));
     } else {
-      // Back out a tendency and apply it.
-      if (m_use_weights) {
-        // get nudging weights field
-        // NOTES: do we really need the vertical interpolation for nudging weights? Since we are going to 
-        //        use the same grids as the case by providing the nudging weights file.
-        //        I would not apply the vertical interpolation here, but it depends...
-        //
-        auto nudging_weights_field = get_helper_field("nudging_weights");
-        // appply the nudging tendencies to the ATM states
-        apply_weighted_tendency(atm_state_field, int_state_field, nudging_weights_field, dt);
-      } else if (m_refine_remap_vert_cutoff > 0.0) {
-        // If we have a cutoff, we apply the tendency with p_mid cutoff
-        // First, get p_mid the field in the atm (i.e., int) state
-        auto p_mid_field = get_field_in("p_mid");
-        // Then, call the tendency with a Heaviside-like cutoff
-        apply_vert_cutoff_tendency(atm_state_field, int_state_field,
-                                   p_mid_field, m_refine_remap_vert_cutoff, dt);
+      vert_interp.setup(team, p_mid_tmp_1d,
+                              ekat::subview(p_tgt,icol));
+    }
+  });
+  Kokkos::fence();
+
+  // Then loop over fields, and do copy_and_pad + vremap
+  auto padded_field = get_helper_field("padded_field");
+  for (const auto& name : m_fields_nudge) {
+    copy_and_pad(get_helper_field(name+"_tmp"),padded_field,false);
+
+    auto field_after_vinterp = get_helper_field(name);
+    auto view_in  = padded_field.get_view<const PackT**>();
+    auto view_out = field_after_vinterp.get_view<PackT**>();
+    // Now use the interpolation object in || over all variables.
+    auto vinterp = KOKKOS_LAMBDA(const MemberType& team) {
+      const int icol = team.league_rank();
+
+      view_1d x1;
+      if (src_pmid_3d) {
+        x1 = ekat::subview(p_mid_tmp_3d,icol);
       } else {
-        apply_tendency(atm_state_field, int_state_field, dt);
+        x1 = p_mid_tmp_1d;
       }
+      auto x2 = ekat::subview(p_tgt,icol);
+
+      auto y1 = ekat::subview(view_in, icol);
+      auto y2 = ekat::subview(view_out,icol);
+
+      vert_interp.lin_interp(team, x1, x2, y1, y2, icol);
+    };
+    Kokkos::parallel_for("nudging_vert_interp_loop", policy_vinterp, vinterp);
+    Kokkos::fence();
+
+    // If timescale==0, the call get_helper_field(name) returns the same
+    // fields as get_field_out_wrap(name) they are alias, so nothing to do.
+    // If timescale>0, then we need to back out a tendency.
+    if (m_timescale > 0) {
+      auto atm_state_field = get_field_out_wrap(name);
+      apply_tendency(atm_state_field,field_after_vinterp,dt);
     }
   }
+  m_atm_logger->info("Running nudging...done!");
```

Comment:
Should this be at the "info" level?  Doesn't that mean it will by default always print to file?

Reply:
Yeah. I can rm it. I may have added it while I was debugging, I don't recall.